client/Elements/Components/TopicConsumer/TopicConsumerList.patternfly.tsx

```diff
@@ -0,0 +1,164 @@
+
+  const onSetPage = (_event, pageNumber: number) => {
+    setPage(pageNumber);
+    setOffset(page * perPage);
+  };
+
+  const onPerPageSelect = (_event, perPage: number) => {
+    setPerPage(perPage);
+  };
+
+  const onClickConsumerGroup = (id, members, partitions, state) => {
+    const consumerDetail = [id, members, partitions, state];
+    setConsumerGroupDetail(consumerDetail);
+    setIsExpanded(true);
+    console.log(isExpanded);
+  };
+
+  const tableColumns = [
+    { title: 'Consumer group ID' },
+    { title: 'Active members for this topic' },
+    { title: 'Unconsumed partitions for this topic' },
+    { title: 'State' },
+  ];
+  const rowData = tableData.map((topicConsumer) => [
+    {
+      title: (
+        <Button
+          variant='link'
+          isInline
+          onClick={() =>
+            onClickConsumerGroup(
+              topicConsumer.id,
+              topicConsumer.members,
+              topicConsumer.partitions,
+              topicConsumer.state
+            )
+          }
+        >
+          {topicConsumer.id}
+        </Button>
+      ),
+    },
+    topicConsumer?.members,
+    topicConsumer?.partitions,
+    {
+      title: (
+        <Label
+          color={topicConsumer.state === 'Stable' ? 'green' : 'red'}
+          icon={
+            topicConsumer.state === 'Stable' ? (
+              <CheckCircleIcon />
+            ) : (
+              <ExclamationCircleIcon />
+            )
+          }
+        >
+          {topicConsumer.state}
+        </Label>
+      ),
+    },
+  ]);
+
+  const actions = [{ title: 'Edit' }, { title: 'Delete' }];
+  return (
+    <>
+      <Card>
+        <Toolbar>
+          <ToolbarContent>
+            <ToolbarItem>
+              <SearchTopicsConsumerList tableData={setTableData} />
+            </ToolbarItem>
+
+            <ToolbarItem variant='pagination'>
+              <Pagination
+                itemCount={rowData.length}
+                perPage={perPage}
+                page={page}
+                onSetPage={onSetPage}
+                widgetId='pagination-options-menu-top'
+                onPerPageSelect={onPerPageSelect}
+              />
+            </ToolbarItem>
+          </ToolbarContent>
+        </Toolbar>
+        <Divider />
+
+        <Table
+          aria-label='Compact Table'
+          variant={TableVariant.compact}
+          cells={tableColumns}
+          rows={
+            page != 1
+              ? rowData.slice(offset, offset + perPage)
+              : rowData.slice(0, perPage)
+          }
+          actions={actions}
+        >
+          <TableHeader />
+          <TableBody />
+        </Table>
+      </Card>
+      {rowData.length > 0 ? (
+        <Pagination
+          itemCount={rowData.length}
+          perPage={perPage}
+          page={page}
+          onSetPage={onSetPage}
+          widgetId='pagination-options-menu-top'
+          onPerPageSelect={onPerPageSelect}
+          offset={0}
```

Comment:
Is the offset here needed?

Reply:
I'm keeping it like this for now, I have started working on creating a common pagination component in the project. Will look into removing this from that component